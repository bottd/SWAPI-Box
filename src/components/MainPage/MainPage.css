@font-face {
  font-family: starWars;
  src: url(https://fonts2u.com/download/starwars.fonthttps://fonts2u.com/download/starwars.TTF) format("opentype");
}

h1 {
  font-family: starWars;
}


