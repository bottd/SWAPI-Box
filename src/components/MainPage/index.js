import React, { Component } from "react";
import PropTypes from "prop-types";
import Favorite from "../Favorite";
import CardContainer from "../CardContainer";
import Buttons from "../Buttons";
import "./MainPage.css";

class MainPage extends Component {
  render() {
    const {
      favoritesCount,
      toggleFavorites,
      updateData,
      handleCardClick,
      selectedCategory,
      //pathUsed,
      cardData,
    } = this.props;
    return (
      <div className="MainPage">
        <h1 className="mainTitle">$ SWAPi-Box $</h1>
        <Favorite
          toggleFavorites={toggleFavorites}
          favoritesCount={favoritesCount}
        />
        <Buttons
          selectedCategory={selectedCategory}
          updateData={updateData}
        />
        <CardContainer
          selectedCategory={selectedCategory}
          cardData={cardData}
          handleCardClick={handleCardClick}
        />
      </div>
    );
  }
}

MainPage.propTypes = {
  favoritesCount: PropTypes.number.isRequired,
  toggleFavorites: PropTypes.func.isRequired,
  updateData: PropTypes.func.isRequired,
  handleCardClick: PropTypes.func.isRequired,
  selectedCategory: PropTypes.string.isRequired,
  //pathUsed: PropTypes.string.isRequired
};

export default MainPage;
