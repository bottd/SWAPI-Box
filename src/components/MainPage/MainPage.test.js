import React from "react";
import MainPage from "./index";
import { shallow } from "enzyme";

describe("MainPage", () => {
  let wrapper = (
    <MainPage
      cardData={[]}
      appFunctionBundle={{}}
      favoritesCount={0}
      selectedCategory={"planets"}
      toggleFavorites={jest.fn()}
      updateData={jest.fn()}
      handleCardClick={jest.fn()}
    />
  );

  it("Renders like snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
