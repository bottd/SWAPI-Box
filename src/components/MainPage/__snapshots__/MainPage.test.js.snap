// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Renders like snapshot 1`] = `
<MainPage
  appFunctionBundle={Object {}}
  cardData={Array []}
  favoritesCount={0}
  handleCardClick={[MockFunction]}
  selectedCategory="planets"
  toggleFavorites={[MockFunction]}
  updateData={[MockFunction]}
/>
`;
